### PR TITLE
Add Sign in with Glow

### PR DESCRIPTION
**Pull request by benborgers**
Adds a Sign in with Glow button in the top right corner. 

| | Glow installed | Glow not installed |
| --- | --- | --- |
| Desktop |  |  |
| Mobile |  |  

## Lingering questions

- [ ] How do I make Typescript happy with using `window.glow`? 
- [ ] How do I persist logged-in state? Currently, when you refresh it goes back to being logged out.

### CODE REVIEW

File: docs/components/UserContext.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        if (window.glow) {
+          setHasGlow(true);
+        }
+      }, 2000);
+    }
+
+    return () => {
+      clearTimeout(timeout);
+    };
+  }, []);
```
Comment (vpontis) [inline docs/components/UserContext.tsx:37]: This actually has some minor issues, if we want to use `setTimeout` we can have a special hook. Much more info in this blog post: https://overreacted.io/making-setinterval-declarative-with-react-hooks/

File: docs/components/shell/TopNav.tsx [hunk 2]
```diff
@@ -50,9 +54,27 @@ export const TopNav = ({
...
+          </Link>
+
+          {userContext && !userContext.user && userContext.hasGlow && (
+            <button onClick={userContext.signIn} className="auth-button">
+              Sign in with Glow
+            </button>
+          )}
+
+          {userContext && !userContext.user && !userContext.hasGlow && (
+            <a
```
Comment (vpontis) [inline docs/components/shell/TopNav.tsx:69]: Let's use the `<GlowSignInButton />` that Pablo made here: https://github.com/luma-team/glow-js/tree/master/packages/glow-react

If that button has some issues, we can fix them. With that button we shouldn't need the `setTimeout` stuff.

**Review by vpontis — CHANGES_REQUESTED** [review]
>  How do I make Typescript happy with using window.glow?

https://github.com/luma-team/glow-js/blob/c204f155d23b37d634454026759e8d143e409c3f/packages/glow-client/src/window-types.ts#L61

```
declare global {
  interface Window {
    glow?: GlowAdapter;
  }
}
```

> How do I persist logged-in state? Currently, when you refresh it goes back to being logged out.

For now use `window.glow.connect()` instead of `signIn` since that will auto sign in the user.

File: docs/components/shell/TopNav.tsx (hunk 2 above)
Comment (benborgers) [reply docs/components/shell/TopNav.tsx:69]: Would it make sense to swap out my `UserProvider` with Pablo’s `GlowProvider` entirely? Or should I keep the custom `UserProvider` and just use that button?

Comment (vpontis) [reply docs/components/shell/TopNav.tsx:69]: Good idea!

**Comment by vpontis** [comment]
Lmk when I should review again when this uses glow-js

**Review by vpontis — APPROVED** [review]
The button behavior is a bit awkward now but we can iterate on that. I want to contribute a bunch of content for the docs by the end of this weekend, then I would like to clean up the docs and prepare them for release next week.

**Comment by benborgers** [comment]
> I also upgraded the dependencies and it looks like this is failing. Can you figure out what's going on?

The `@glow-app/glow-react` package has dependencies on `react` and `react-dom`, and so does `docs/`, so when those versions don’t match it fails. I think the fix is to [make `react` and `react-dom` a peer dependency and a dev dependency](https://stackoverflow.com/a/30454133). I’m trying to confirm that but haven’t been able to get `npm link` between `glow-react` and `nftoken/docs` working to test that out.

**Comment by vpontis** [comment]
> > I also upgraded the dependencies and it looks like this is failing. Can you figure out what's going on?
> 
> 
> 
> The `@glow-app/glow-react` package has dependencies on `react` and `react-dom`, and so does `docs/`, so when those versions don’t match it fails. I think the fix is to [make `react` and `react-dom` a peer dependency and a dev dependency](https://stackoverflow.com/a/30454133). I’m trying to confirm that but haven’t been able to get `npm link` between `glow-react` and `nftoken/docs` working to test that out. 

Ah gotcha. Peer dep sounds good. Want to open a PR in glow-react?

**Review by vpontis — APPROVED** [review]

**Pull request merged**